tests/system_tests/backend_fixtures.py

```diff
@@ -0,0 +1,215 @@
+
+
+@dataclass(frozen=True)
+@dataclass_transform(frozen_default=True)
+class FixtureCmd:
+    path: ClassVar[str]  # e.g. "db/user"
+
+    command: str
+
+
+@dataclass(frozen=True)
+class UserCmd(FixtureCmd):
+    path: ClassVar[str] = "db/user"
+
+    command: Literal["up", "down", "down_all", "logout", "login", "password"]
+
+    username: str | None = None
+    password: str | None = None
+    admin: bool = False
+
+
+def random_string(alphabet: str = ascii_lowercase + digits, length: int = 12) -> str:
+    """Generate a random string of a given length.
+
+    Args:
+        alphabet: A sequence of allowed characters in the generated string.
+        length: Length of the string to generate.
+
+    Returns:
+        A random string.
+    """
+    return "".join(secrets.choice(alphabet) for _ in range(length))
+
+
+@dataclass
+class BackendFixtureFactory:
+    service_url: InitVar[str]  #: Local base URL for backend fixture service.
+
+    worker_id: str  #: Identifies the current worker in pytest-xdist parallel runs.
+
+    _client: httpx.Client = field(init=False)
+    _cleanup_stack: deque[FixtureCmd] = field(default_factory=deque, init=False)
+
+    def __post_init__(self, service_url: str):
+        self._client = httpx.Client(
+            base_url=service_url,
+            # event_hooks={"response": [httpx.Response.raise_for_status]},
+            # timeout=None,
+        )
+
+    def __enter__(self):
+        self._client.__enter__()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.cleanup()
+        self._client.__exit__()
+
+    def make_user(self, name: str | None = None, admin: bool = False) -> str:
+        """Create a new user and return their username."""
+        name = name or f"user-{self.worker_id}-{random_string()}"
+
+        self.send_cmds(
+            UserCmd("up", username=name, admin=admin),
+            UserCmd("password", username=name, password=name, admin=admin),
+        )
+
+        # Register command(s) to delete the user on cleanup
+        self._cleanup_stack.append(
+            UserCmd("down", username=name, admin=admin),
+        )
+        return name
+
+    def send_cmds(self, *cmds: FixtureCmd) -> None:
+        for cmd in cmds:
+            self._send(cmd.path, data=asdict(cmd))
+
+    def _send(self, path: str, data: dict[str, Any]) -> None:
+        # trigger fixture
+        endpoint = str(self._client.base_url.join(path))
+        # FIXME: Figure out how SDK team preferences/conventions for replacing print statements
+        print(f"Triggering fixture on {endpoint!r}: {data!r}", file=sys.stderr)  # noqa: T201
+        try:
+            response = self._client.post(path, json=data)
+            response.raise_for_status()
+        except httpx.HTTPStatusError as e:
+            # FIXME: Figure out how SDK team preferences/conventions for replacing print statements
+            print(e.response.json(), file=sys.stderr)  # noqa: T201
+
```

Comment:
fixmes here

Reply:
Right thanks for the reminder -- 🙏

`print` statement was kept from existing code, but are there SDK team preferences for what to replace it with?  (Eg if there are logging/reporting patterns used elsewhere)